server/src/main/java/org/candlepin/resource/HypervisorResource.java

```diff
@@ -186,6 +190,47 @@ else if (guestIdsUpdated) {
         return result;
     }
 
+    /**
+     * Creates or Updates the list of Hypervisor hosts
+     * <p>
+     * Allows agents such as virt-who to update hosts' information . This is typically
+     * used when a host is unable to register to candlepin via subscription manager.
+     * <p>
+     * In situations where consumers already exist it is probably best not to
+     * allow creation of new hypervisor consumers.  Most consumers do not
+     * have a hypervisorId attribute, so that should be added manually
+     * when necessary by the management environment.
+     *
+     * @param hypervisorJson the json representation of the hypervisors
+     * @param principal
+     * @param ownerKey key of owner to update
+     * @param createMissing specify whether or not to create missing hypervisors.
+     * Default is true.  If false is specified, hypervisorIds that are not found
+     * will result in a failed state of the job.
+     * @return a JobDetail object
+     *
+     * @httpcode 202
+     * @httpcode 200
+     *
+     */
+    @POST
+    @Consumes(MediaType.TEXT_PLAIN)
+    @Produces(MediaType.APPLICATION_JSON)
+    @Transactional
+    @Path("/{owner}")
+    public JobDetail updateOwnersHypervisors(
+        String hypervisorJson, @Context Principal principal,
+        @PathParam("owner") @Verify(value = Owner.class,
+            require = Access.READ_ONLY,
+            subResource = SubResource.HYPERVISOR) String ownerKey,
+        @QueryParam("create_missing") @DefaultValue("true") boolean createMissing) {
+
+        log.info("Hypervisor update by principal: " + principal);
+        Owner owner = this.getOwner(ownerKey);
+
+        return HypervisorUpdateJob.forOwner(owner, hypervisorJson, createMissing, principal);
+    }
```

Comment:
We need some way for virt-who to check which API call it can make, typically I think we're using GET / and whether or not that container something we're interested in. A capability might also make sense but we don't do that yet afaik.